test/framework/toolchain.py

```diff
@@ -338,6 +352,30 @@ def test_optarch_generic(self):
                         else:
                             self.assertFalse(generic_flags in tc.get_variable(var))
 
+    def test_optarch_aarch64_heuristic(self):
+        """Test whether AArch64 pre-GCC-6 optimal architecture flag heuristic works."""
+        st.get_cpu_architecture = lambda: st.AARCH64
+        st.get_cpu_model = lambda: 'ARM Cortex-A53'
+        st.get_cpu_vendor = lambda: st.ARM
+        tc = self.get_toolchain("GCC", version="4.7.2")
+        tc.set_options({})
+        tc.prepare()
+        self.assertEqual(tc.options.options_map['optarch'], 'mcpu=cortex-a53')
+        self.assertTrue('-mcpu=cortex-a53' in os.environ['CFLAGS'])
+
+        tc = self.get_toolchain("GCCcore", version="6.2.0")
+        tc.set_options({})
+        tc.prepare()
+        self.assertEqual(tc.COMPILER_OPTIMAL_ARCHITECTURE_OPTION[tc.arch], 'mcpu=native')
```

Comment:
Is it OK to compare with `tc.COMPILER_OPTIMAL_ARCHITECTURE_OPTION` here?  In the other cases we compare against `tc.options.options_map['optarch']`.